# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/PCanvas.ipdl
# Commit: 31935f164ef0
# Full Hash: 31935f164ef0d6c4668f211482c139847dcab7f2
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2023-12-13 00:43:26
# Regressor Bug: 1829026
# File Overlap Count: 1
# Description:
#   Bug 1829026 - Update CanvasTranslator to work with DrawTargetWebgl. r=aosmond
#   
#   This adds the necessary infrastructure for CanvasTranslator to allocate DrawTargetWebgl
#   instead of just allocating TextureData, and to use RemoteTextureMap to handle sending
#   the DrawTargetWebgl frames to the compositor.
# ==============================================================================

diff -r 13c9bf69085f -r 31935f164ef0 gfx/layers/ipc/PCanvas.ipdl
--- a/gfx/layers/ipc/PCanvas.ipdl	Tue Dec 12 07:35:02 2023 +0000
+++ b/gfx/layers/ipc/PCanvas.ipdl	Tue Dec 12 07:35:03 2023 +0000
@@ -11,6 +11,7 @@
 [MoveOnly] using mozilla::CrossProcessSemaphoreHandle from "mozilla/ipc/CrossProcessSemaphore.h";
 using mozilla::layers::TextureType from "mozilla/layers/LayersTypes.h";
 [MoveOnly] using mozilla::ipc::SharedMemoryBasic::Handle from "mozilla/ipc/SharedMemoryBasic.h";
+using mozilla::gfx::BackendType from "mozilla/gfx/Types.h";
 
 namespace mozilla {
 namespace layers {
@@ -33,8 +34,9 @@
    * aUseIPDLThread if true, use the IPDL thread instead of the worker pool for
    * translation requests
    */
-  async InitTranslator(TextureType aTextureType, Handle aHeaderHandle,
-                       Handle[] aBufferHandles, uint64_t aBufferSize,
+  async InitTranslator(TextureType aTextureType, BackendType aBackendType,
+                       Handle aHeaderHandle, Handle[] aBufferHandles,
+                       uint64_t aBufferSize,
                        CrossProcessSemaphoreHandle aReaderSem,
                        CrossProcessSemaphoreHandle aWriterSem,
                        bool aUseIPDLThread);
@@ -67,6 +69,22 @@
    * Deactivate remote canvas, which will cause fall back to software.
    */
   async Deactivate();
+
+  /**
+   * Block further accelerated canvases from being created, but allow existing
+   * canvases to continue processing.
+   */
+  async BlockCanvas();
+
+  /**
+   * Notify that a remote accelerated canvas requires a fallback refresh.
+   */
+  async NotifyRequiresRefresh(int64_t aTextureId);
+
+  /**
+   * Cache the shmem of the framebuffer for snapshotting.
+   */
+  async SnapshotShmem(int64_t aTextureId, Shmem aShmem) returns (bool aSuccess);
 };
 
 } // layers